packages/node/src/indexer/indexer.manager.ts

```diff
@@ -340,14 +340,29 @@ export class IndexerManager {
       } else {
         await metadataRepo.upsert({ key: 'genesisHash', value: genesisHash });
       }
+    } else {
+      assert(
+        this.project.network === keyValue.genesisHash,
+        'Specified project manifest genesis hash does not match database stored genesis hash, consider cleaning project schema using --force-clean',
+      );
     }
 
     if (keyValue.chain !== chain) {
       await metadataRepo.upsert({ key: 'chain', value: chain });
+    } else {
+      assert(
+        chain === keyValue.chain,
```

Comment:
Only need to do the check for genesisHash, because chain and specName could be different after a chain upgrade.